src/ai/backend/agent/docker/agent.py

```diff
@@ -1487,7 +1487,13 @@ async def push_image(self, image_ref: ImageRef, registry_conf: ImageRegistry) ->
             }
 
         async with closing_async(Docker()) as docker:
-            await docker.images.push(image_ref.canonical, auth=auth_config)
+            # TODO: Remove this useless type casting after resolving https://github.com/aio-libs/aiodocker/pull/909.
+            result = cast(list, await docker.images.push(image_ref.canonical, auth=auth_config))
+
+            if not result:
+                raise RuntimeError("Failed to push image: unknown error")
```

Comment:
It seems better to indicate that an unexpected result was returned rather than an unknown error.